src/irr/asset/CGLTFLoader.h

```diff
@@ -58,29 +58,34 @@ namespace irr
 					SGLTFPT_TRIANGLE_FAN
 				};
 
+				/*
+					A node in the node hierarchy. A node can have either a matrix or any combination of 
+					translation/rotation/scale (TRS) properties.
+				*/
+
 				struct SGLTFNode
 				{
-					std::optional<std::string> name;
-					std::optional<std::vector<uint32_t>> children;
-					std::optional<uint32_t> mesh;
-					std::optional<uint32_t> camera;
-					std::optional<uint32_t> skin;
-					std::optional<uint32_t> weights;
+					std::optional<std::string> name;	//!< The user-defined name of this object.
+					std::optional<std::vector<uint32_t>> children;	//!< The indices of this node's children.
+					std::optional<uint32_t> mesh;		//!< The index of the mesh in this node.
+					std::optional<uint32_t> camera;		//!< The index of the camera referenced by this node.
+					std::optional<uint32_t> skin;		//!< The index of the skin referenced by this node.
+					std::optional<uint32_t> weights;	//!< The weights of the instantiated Morph Target. Number of elements must match number of Morph Targets of used mesh.
 
 					struct SGLTFNTransformationTRS
 					{
-						core::vector3df_SIMD translation;	//!< translation, local coordinate system
-						core::vector3df_SIMD scale;			//!< scale, local coordinate system
-						core::vector4df_SIMD rotation;		//!< quaternion, local coordinate system
+						core::vector3df_SIMD translation;	//!< The node's translation along the x, y, and z axes.
+						core::vector3df_SIMD scale;			//!< The node's non-uniform scale, given as the scaling factors along the x, y, and z axes.
+						core::vector4df_SIMD rotation;		//!< The node's unit quaternion rotation in the order (x, y, z, w), where w is the scalar.
 					};
 
 					union SGLTFTransformation
 					{
 						SGLTFTransformation() {}
 						~SGLTFTransformation() {}
 
-						SGLTFNTransformationTRS trs;
-						core::matrix4SIMD matrix;
+						SGLTFNTransformationTRS trs;	
+						core::matrix4SIMD matrix;		//!< A floating-point 4x4 transformation matrix stored in column-major order.
```

Comment:
why 4x4, a TRS matrix is representable in 4x3/3x4 entirely

Reply:
solved